src/pymmcore_widgets/_mda/_mda_widget.py

```diff
@@ -308,6 +308,9 @@ def set_state(self, state: dict | MDASequence | str | Path) -> None:
             MDASequence state in the form of a dict, MDASequence object, or a str or
             Path pointing to a sequence.yaml file
         """
+        # TODO: prevent _update_total_time from being called until
+        # all subcomponents have been set
```

Comment:
@tlambert03 any suggestions here? I looked at https://pyapp-kit.github.io/superqt/utilities/throttling/#superqt.utils._throttler.GenericSignalThrottler.throttle but that didn't seem right as the signals are already connected to the existing function. We also can't block them in case something else is listening, it seems like we need to block a specific function from receiving the signals

Reply:
yeah, a debouncer would be a good idea here.  They would need to be debounced at connection time though, so it would look something like 

```py
from superqt.utils import qdebounced

self.channel_widget.valueChanged.connect(qdebounced(self._update_total_time))
```

but of course, that's hooked up a million times... so it would need to go everywhere.  can follow up in a later PR